tough-kms/src/error.rs

```diff
@@ -0,0 +1,121 @@
+
+use snafu::{Backtrace, Snafu};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// The error type for this library.
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(crate)")]
+pub enum Error {
+    #[snafu(display(
+    "Failed to create customer managed key for aws-kms://{}/{}: {}",
+    profile.as_deref().unwrap_or(""),
+    key_id,
+    source,
+    ))]
+    KmsCreateKey {
+        profile: Option<String>,
+        key_id: String,
+        source: rusoto_core::RusotoError<rusoto_kms::CreateKeyError>,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display(
+    "Failed to assign alias {} for created customer managed key_id {} for aws-kms://{}/{}: {}",
+    alias,
+    target_key_id,
+    profile.as_deref().unwrap_or(""),
+    key_id,
+    source,
+    ))]
+    KmsCreateAlias {
+        alias: String,
+        target_key_id: String,
+        profile: Option<String>,
+        key_id: String,
+        source: rusoto_core::RusotoError<rusoto_kms::CreateAliasError>,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Error creating AWS credentials provider: {}", source))]
+    RusotoCreds {
+        source: rusoto_credential::CredentialsError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Unable to determine region from profile '{}': {}", profile, source))]
+    RusotoRegionFromProfile {
+        profile: String,
+        source: rusoto_credential::CredentialsError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Unknown AWS region \"{}\": {}", region, source))]
+    RusotoRegion {
+        region: String,
+        source: rusoto_core::region::ParseRegionError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Error creating AWS request dispatcher: {}", source))]
+    RusotoTls {
+        source: rusoto_core::request::TlsError,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Unable to create tokio runtime: {}", source))]
+    RuntimeCreation {
+        source: std::io::Error,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display(
+    "Failed to get public key for aws-kms://{}/{} : {}",
+    profile.as_deref().unwrap_or(""),
+    key_id,
+    source,
+    ))]
+    KmsGetPublicKey {
+        profile: Option<String>,
+        key_id: String,
+        source: rusoto_core::RusotoError<rusoto_kms::GetPublicKeyError>,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display(
+    "Alias {} exist for aws-kms:://{}/{}",
+    key_id,
+    profile.as_deref().unwrap_or(""),
+    key_id
+    ))]
+    KmsAliasCheck {
+        profile: Option<String>,
+        key_id: String,
+    },
+
+    #[snafu(display("Public key is none"))]
+    PublicKeyNone,
+
+    #[snafu(display(
+    "Write in Not defined for aws-kms://{}/{}",
+    profile.as_deref().unwrap_or(""),
+    key_id
+    ))]
+    WriteNotDefined {
+        profile: Option<String>,
+        key_id: String,
+    },
+
+    #[snafu(display(
+    "Key metatada is None while creating key for aws-kms://{}/{}",
```

Comment:
Rather than referring to Rust concepts of `Some() / None` in these error messages, let's describe the problem in prose. Are we trying to inform the user that the metadata is empty? doesn't exist?